lib/data/external/utils.js

```diff
@@ -20,14 +20,14 @@ const backendHealth = {
 /* eslint-enable camelcase */
 
 const utils = {
-    logHelper(log, level, description, error, dataStoreName) {
+    logHelper(log, level, description, error, dataStoreName, backendType) {
         log[level](description, { error: error.message,
-          errorName: error.name, dataStoreName });
+          errorName: error.name, dataStoreName, backendType });
     },
     // take off the 'x-amz-meta-'
     trimXMetaPrefix(obj) {
         const newObj = {};
-        Object.keys(obj).forEach(key => {
+        Object.keys(obj || {}).forEach(key => {
```

Comment:
This is a bit hard to follow. Perhaps this could be made more explicit by adding a separate condition for the case which this addresses.